app/scripts/inpage.js

```diff
@@ -38,8 +37,24 @@ web3.setProvider = function () {
   log.debug('MetaMask - overrode web3.setProvider')
 }
 log.debug('MetaMask - injected web3')
-// export global web3, with usage-detection
-setupDappAutoReload(web3, inpageProvider.publicConfigStore)
+
+// export global web3, with usage-detection and deprecation warning
+let hasBeenWarned = false
+global.web3 = new Proxy(web3, {
```

Comment:
Reasonable place to move it, I assume you've manually verified the warning still works?

Reply:
Yes, manually verified in Chrome.